components/dialog/stories/template.js

```diff
@@ -24,65 +24,86 @@ export const Template = ({
 	footer = [],
 	customClasses = [],
 	id = getRandomId("dialog"),
+	size = "medium",
+	layout,
+	hasHeroImage = false,
+	heroImageUrl,
+	customStyles = {},
 } = {}, context = {}) => {
-	const { globals = {}, updateArgs } = context;
-	const scale = globals.scale ?? "medium";
-	const toggleOpen = function () {
-		updateArgs({ isOpen: !isOpen });
-	};
+	const { updateArgs } = context;
+	const toggleOpen = () => updateArgs({ isOpen: !isOpen });
 
 	const Dialog = html`
 		<div
 			class=${classMap({
 				[rootClass]: true,
-				[`${rootClass}--${scale}`]: true,
-				[`${rootClass}--dismissable`]: isDismissable && !isFullScreen && !isFullScreenTakeover,
-				[`${rootClass}--fullscreen`]: isFullScreen,
-				[`${rootClass}--fullscreenTakeover`]: isFullScreenTakeover,
+				[`${rootClass}--dismissable`]: isDismissible && (layout !== "fullscreen" || layout !== "fullscreenTakeover"),
+				[`${rootClass}--${layout}`]: typeof layout !== "undefined" && layout !== "default",
```

Comment:
Maybe we just remove default and the absence of layout sets the class since default doesn't use a class.

```suggestion
				[`${rootClass}--${layout}`]: typeof layout !== "undefined",
```